src/StringTools.cpp

```diff
@@ -0,0 +1,617 @@
+        std::transform(Begin,End,Begin,[](int ToConvert) -> char {
+            return static_cast<char>( ::toupper(ToConvert) );
+        });
+    }
+
+    String UpperCaseCopy(String Source)
+    {
+        StringTools::ToUpperCase(Source.begin(),Source.end());
+        return Source;
+    }
+
+    void ToLowerCase(StrIter Begin, StrIter End)
+    {
+        std::transform(Begin,End,Begin,[](int ToConvert) -> char {
+            return static_cast<char>( ::tolower(ToConvert) );
+        });
+    }
+
+    String LowerCaseCopy(String Source)
+    {
+        StringTools::ToLowerCase(Source.begin(),Source.end());
+        return Source;
+    }
+
+    void ToCamelCase(StrIter Begin, StrIter End)
+    {
+        Boole PrevCharIsWhitespace = true;
+        for( StrIter CurrIt = Begin ; CurrIt != End ; ++CurrIt )
+        {
+            Boole CurrCharIsLowerLetter = StringTools::IsLowerAlphaLetter( *CurrIt );
+            Boole CurrCharIsUpperLetter = StringTools::IsUpperAlphaLetter( *CurrIt );
+            if( PrevCharIsWhitespace && CurrCharIsLowerLetter ) {
+                (*CurrIt) = static_cast<Char8>( ::toupper(*CurrIt) );
+            }else if( !PrevCharIsWhitespace && CurrCharIsUpperLetter ) {
+                (*CurrIt) = static_cast<Char8>( ::tolower(*CurrIt) );
+            }
+            PrevCharIsWhitespace = StringTools::IsWhitespace( *CurrIt );
+        }
+    }
+
+    String CamelCaseCopy(String Source)
+    {
+        StringTools::ToCamelCase(Source.begin(),Source.end());
+        return Source;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Size Modifying String Manipulations
+
+    void Trim(String& Source, const Boole Left, const Boole Right)
+    {
+        const String Delims = " \t\r";
+        if( Left ) {
+            size_t Forward = 0;
+            while( Forward < Source.size() )
+            {
+                if( Delims.find( Source[Forward] ) == String::npos ) {
+                    break;
+                }
+                ++Forward;
+            }
+            Source.erase(0,Forward);
+        }
+        if( Right ) {
+            size_t Reverse = Source.size();
+            while( Reverse > 0 )
+            {
+                if( Delims.find( Source[Reverse-1] ) == String::npos ) {
+                    break;
+                }
+                --Reverse;
+            }
+            Source.erase(Reverse);
+        }
+    }
+
+    String TrimCopy(String Source, const Boole Left, const Boole Right)
+    {
+        StringTools::Trim(Source,Left,Right);
+        return Source;
+    }
+
+    void RemoveDuplicateWhitespaces(String& Source)
+    {
+        static const char* Delims = " \t";
+        for( size_t CurrIndex = Source.find_first_of(Delims,0,2) ;
+             CurrIndex != String::npos ;
+             CurrIndex = Source.find_first_of(Delims,CurrIndex,2) )
+        {
+            size_t EndIndex = CurrIndex;
+            Char8 CurrWhite = Source[EndIndex];
```

Comment:
Would this work on a string with several alternating tabs and spaces in between words?

Reply:
No, nor was it meant to when it was written.  It could be updated to handle such cases...although I'm curious what the use case for that would be and which of the alternating whitespace characters should be prioritized when replacing the sequence with one character.